lib/utils/execution-state-manager.js

```diff
@@ -0,0 +1,194 @@
+'use strict';
+
+/**
+ * A class to store the state of an execution.
+ *
+ * @class ExecutionStateManager
+ */
+class ExecutionStateManager {
+  constructor() {
+    this.initializeStates();
+  }
+
+  initializeStates() {
+    // A map of browerId to test modules executed on that browser read from test-execution.json.
+    this._replayExecutionMap = null;
+
+    // A map of browerId to test modules executed for the current test execution.
+    this._browserToModuleMap = Object.create(null);
```

Comment:
Can we use a `Map` here?

Reply:
Same thing here. We could, but do you see any difference between using `Map` vs `Object.create(null)` in this case?

Reply:
IMO it's readability, as `Object.create(null)` is less immediately intuitive that you're wanting `Map` functionality. With a `Map` I think it's pretty clear. Thoughts?

Reply:
with a suggestion from @stefanpenner https://github.com/ember-cli/ember-exam/pull/136#discussion_r239592959 `Object.create(null)` is used here. what do you think which brings more benefits?

Reply:
After off-line discussion, decided to change to use a `Map` as  a `Map` has less edge cases and  it gives more readability when neither gives no significant performance issues.